### PR TITLE
added loading button and loader

### PR DESCRIPTION
**Pull request by JantuDeb**
# Loader
- added loading component for video listing page.
- added loading in button.
- refactored some codes.
- added error and loading state in reducer

### CODE REVIEW

**Review by munjalSaksham71 — APPROVED** [review]
Loader Works absolutely fine and its looking good. good Work!   If you have completed everything - I would love to suggest you that add lazy loader - that would optimize your app performance a bit.

File: src/components/loader/Button.css [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/* @keyframes spinner {
+    0% {
+        transform: rotate(0deg);
+    }
+
+    100% {
+        transform: rotate(360deg);
+    }
+} */
```
Comment (mariyasada) [inline src/components/loader/Button.css:29]: remove this if not necessary

**Review by mariyasada — APPROVED** [review]
the loader works completely fine. Liked that you are using notes here.

File: src/components/loader/Loader.css [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+
+@keyframes spinner {
+    0% {
+        transform: rotate(0deg);
+    }
+
+    100% {
+        transform: rotate(360deg);
```
Comment (ajit-sharmaGH) [inline src/components/loader/Loader.css:104]: Love this rotational geometry!

**Review by ajit-sharmaGH — APPROVED** [review]

File: src/components/loader/Button.css (hunk 1 above)
Comment (JantuDeb) [reply src/components/loader/Button.css:29]: Will removed

**Pull request merged**